strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -537,6 +537,10 @@ def fidelity_coherent(self, alpha, modes=None):
             / np.sqrt(np.linalg.det(cov_sum))
         )
         fidelity = np.sum(weighted_exp)
+
+        # Numerical error can yield fidelity marginally greater than 1
```

Comment:
I had to add this in because I was sometimes getting vacuum fidelity of 1+machine_eps, and then that meant a negative probability in the threshold sampling for 1-vacuum_fidelity.